k2/csrc/ragged_ops_inl.h

```diff
@@ -566,24 +568,63 @@ void ArgMaxPerSublist(Ragged<T> &src, T initial_value, Array1<int32_t> *dst) {
     }
   } else {
     K2_CHECK_EQ(c->GetDeviceType(), kCuda);
-    argmax_internal::PairInputIterator<T> input_iter(values_data);
```

Comment:
Shall we remove everything inside the namespace `argmax_internal`?

Reply:
Since it's not used now I guess it's OK to remove it, we can always restore it from the git history if we need.
It still bothers me a little that we haven't found the bug though.

Reply:
> It still bothers me a little that we haven't found the bug though.


I just found it!

The following implementation is buggy:
https://github.com/k2-fsa/k2/blob/818b138b33eabe440601df8910a2b97ac088594b/k2/csrc/ragged_ops_inl.h#L469-L471

`t_[idx]` should be changed to `t_[idx + offset_]`

Reply:
Oops !